### PR TITLE
Update README to link to mattermost.com 

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
There's a number of links in the web app's README that are outdated and don't match the server's README, so I've updated the ones that point to mattermost.org. There's a few links to pages that don't have a non-mattermost.org equivalent, so I've left those as-is.

Hopefully this'll help further reduce mattermost.org's SEO so that someone on the website side will finally remove it 😛.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by hmhealey** [comment]
@cwarnermm I'm not sure if the READMEs fall into your team's scope. We don't make changes to them frequently, but let me know if you want to be assigned to these if they do come up.

**Comment by cwarnermm** [comment]
Thanks, @hmhealey! README docs for repos may not formally fall under the Docs Team's scope, we're happy to help with updates. Please feel free to continue including me as an editor on PRs going forward, or log Jira tickets and assign them to me if reviews or substantial rewrites are needed.

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Install on your own machine:
 
 - [One-line Docker Preview](http://docs.mattermost.com/install/docker-local-machine.html#one-line-docker-install) 
 - [Developer Machine Setup](https://docs.mattermost.com/developer/dev-setup.html)
```
Comment (cwarnermm) [inline README.md:27]: This URL can be changed to: https://developers.mattermost.com/contribute/server/developer-setup/

**Review by cwarnermm — COMMENTED** [review]
Updated URLs for docs resources provided inline.

**Review by esethna — APPROVED** [review]

**Review by cwarnermm — APPROVED** [review]
Thanks!

**Pull request merged**